src/interpreter/InterpretOneOfWithProperties.ts

```diff
@@ -0,0 +1,46 @@
+import { CommonModel } from '../models/CommonModel';
+import { Interpreter, InterpreterOptions, InterpreterSchemaType } from './Interpreter';
+
+/**
+ * Interpreter function for oneOf keyword with properties.
+ * 
+ * It puts the schema reference into the items field.
```

Comment:
Do you mind clarifying a bit more in-depth the pattern it matches and how it's interpreted? 

Cause if we don't look at this for the next couple of months, it might be a bit troubling to get an overview 😅